src/examples/chord/chord.ts

```diff
@@ -0,0 +1,240 @@
+
+interface ChordVisualization extends VisualizationDefinition {
+  svg?: any,
+  tooltip?: any,
+  computeMatrix: (data: VisData, dimensions: string[], measure: string) => any,
+  titleText: (lookup: any, source: any, target: any, formatter: any) => string,
+}
+
+const vis: ChordVisualization = {
+  id: 'chord',
+  label: 'Chord',
+  options: {
+    color_range: {
+      type: 'array',
+      label: 'Color Range',
+      display: 'colors',
+      default: ['#dd3333', '#80ce5d', '#f78131', '#369dc1', '#c572d3', '#36c1b3', '#b57052', '#ed69af']
+    }
+  },
+  // Set up the initial state of the visualization
+  create(element, config) {
+    element.innerHTML = `
+      <style>
+        .chordchart circle {
+          fill: none;
+          pointer-events: all;
+        }
+
+        .chordchart:hover path.chord-fade {
+          display: none;
+        }
+
+        .groups text {
+          font-size: 12px;
+        }
+
+        .chord-tip {
+          position: absolute;
+          top: 0;
+          left: 0;
+          z-index: 10;
+        }
+      </style>
+    `
+
+    this.tooltip = d3.select(element).append('div').attr('class', 'chord-tip')
+
+    this.svg = d3.select(element).append('svg')
+  },
+
+  computeMatrix(data, dimensions, measure) {
+    const indexByName = d3.map()
+    const nameByIndex = d3.map()
+    const matrix: any[] = []
+    let n = 0
+
+    // Compute a unique index for each package name.
+    dimensions.forEach((dimension) => {
+      data.forEach((d) => {
+        const value = d[dimension].value
+        if (!indexByName.has(value)) {
+          nameByIndex.set(n.toString(), value)
+          indexByName.set(value, n++)
+        }
+      })
+    })
+
+    // Construct a square matrix
+    for (let i = -1; ++i < n;) {
+      matrix[i] = []
+      for (let t = -1; ++t < n;) {
+        matrix[i][t] = 0
+      }
+    }
+
+    // Fill matrix
+    data.forEach(function(d) {
+      const row = indexByName.get(d[dimensions[1]].value)
+      const col = indexByName.get(d[dimensions[0]].value)
+      const val = d[measure].value
+      matrix[row][col] = val
+    })
+
+    return {
+      matrix,
+      indexByName,
+      nameByIndex
+    }
+  },
+
+  // Render in response to the data or settings changing
+  update(data, element, config, queryResponse) {
+    if (!handleErrors(this, queryResponse, {
+      min_pivots: 0, max_pivots: 0,
+      min_dimensions: 2, max_dimensions: 2,
+      min_measures: 1, max_measures: 1
+    })) return
+
+    const dimensions = queryResponse.fields.dimension_like
+    const measure = queryResponse.fields.measure_like[0]
+
+    // Set dimensions
+    const width = element.clientWidth
+    const height = element.clientHeight
+    const margin = 10
+    const thickness = 15
+    const outerRadius = Math.min(width, height) * 0.5
+    const innerRadius = outerRadius - thickness
+
+    // Stop if radius is < 0
+    // TODO: show warning to user ???
+    // TODO: Set a min-radius ???
```

Comment:
Possibly remove TODOs?

Reply:
@nickfogler these TODOs were pre-existing in the JS version. should i keep them in or remove them?